packages/utils/src/carequality/download-directory.ts

```diff
@@ -70,13 +72,24 @@ async function main() {
   const startedAt = Date.now();
   console.log(`############## Started at ${new Date(startedAt).toISOString()} ##############`);
   initRunsFolder(dirName);
+  const outputFilename = getFolderName() + ".json";
 
   const orgs = await downloadCQDirectory(apiKey, apiMode);
+  console.log(`Downloaded ${orgs.length} organizations, storing them on file ${outputFilename}`);
+
+  const chunks = chunk(orgs, 1_000);
+  let idx = 0;
+  fs.appendFileSync(outputFilename, "{");
+  for (const chunk of chunks) {
+    const chunkProp = { [`chunk-${++idx}`]: chunk };
+    const chunkString = JSON.stringify(chunkProp, null, 2);
+    const chunkContents = chunkString.slice(1, -1);
```

Comment:
we could prob. run into problems if there are any escape characters -- leading to invalid json.

Maybe we could use streaming here?

```suggestion
    import { createWriteStream } from 'fs';
    ...
    
    stream.write(`  "chunk-${++idx}": `);
    stream.write(JSON.stringify(chunk)); 
    
    if (idx < totalChunks) {
        stream.write(',\n');
      } else {
        stream.write('\n');
      }
```

Reply:
The return from the API is JSON, so it should be well formed - we parse it as such in the CQ directory rebuild logic.

This is just a util to store the dir on local in a very rare, exceptional situation we want to check the contents before it gets converted into our DB representation.

Lastly, how would streaming help us here?

Reply:
basically for better memory management...

Reply:
Yeah, for memory management streaming is a great option. But this is just a util script we very rarely use, so I'll leave as is - it doesn't justify invest more time into this, the main thing being testing it properly.

Reply:
And BTW, at this point the objects are already stored in memory, would need to change `downloadCQDirectory`, which is used in prod and that would require much more work and testing.